rollover.py

```diff
@@ -245,6 +230,24 @@ def main():
         print "done"
 
         #
+        # Query services and tasks just before calling
+        #
+        # NOTE: If a deployment is made and scheduled to the machine being
+        # removed after the services and tasks are queried, but before
+        # deregister_container_instance() is called, then it wont be tracked
+        # and removed during the rollover. The following calls are grouped
+        # together as closely as possible to minimize this risk.
+        #
```

Comment:
Crazy Idea: Might we add the notion of a "pause" or "deploy lock" in catapult? This would mean we could pause all deploys temporarily, so we can be confident maintenance operations are safe.

Either 

(1) Deploy operations would still be received by Catapult, and if it sees deploys are locked it will wait to carry out the deploy until lock is removed. 
(2) Deploys are immediately rejected while the lock is enabled.

--

We might be able to design this more flexibly to cover another use case. Often people will say something like "don't deploy oauth!" Could we encode this, such that the user gets feedback that a deploy block is suggested/required globally/on an app?